lib/strategies/persistent.js

```diff
@@ -25,21 +25,45 @@ module.exports = {
   processString: function(ctx, contents, relativePath) {
     var key = ctx.cacheKeyProcessString(contents, relativePath);
     var cache = this._cache;
+    var string;
 
     return cache.get(key).then(function(entry) {
       if (entry.isCached) {
         ctx._debug('persistent cache hit: %s', relativePath);
-        return entry.value;
+
+        string = Promise.resolve(entry.value)
+          .then(function(value) {
+            return JSON.parse(value);
+          });
       } else {
         ctx._debug('persistent cache prime: %s', relativePath);
-        var string = Promise.resolve(ctx.processString(contents, relativePath));
+        string = Promise.resolve(ctx.processString(contents, relativePath))
+          .then(function(result) {
+            if (typeof result === 'string') {
+              return { output: result };
+            } else {
+              return result;
+            }
+          })
+          .then(function(value) {
+            var stringValue = JSON.stringify(value);
 
-        return string.then(function(string) {
-          return cache.set(key, string).then(function() {
-            return string;
+            return cache.set(key, stringValue).then(function() {
+              return value;
+            });
           });
-        });
+
       }
+
+      return string
+        .then(function(object) {
+          var output = object.output;
+
+          return Promise.resolve(ctx.postProcess(object, relativePath))
+            .then(function() {
+              return output;
```

Comment:
i think its ok for postProcess to intercept and provide a new return value. (unless you don't think so)